src/devices_models/devices/common/cost_functions.jl

```diff
@@ -59,10 +70,12 @@ function cost_function!(
     ::Type{<:PM.AbstractPowerModel},
     feedforward::Union{Nothing, AbstractAffectFeedForward} = nothing,
 ) where {T <: PSY.Component, U <: AbstractDeviceFormulation}
-    spec = AddCostSpec(T, U, psi_container)
-    @debug T, spec
     for d in devices
-        add_to_cost!(psi_container, spec, PSY.get_operation_cost(d), PSY.get_name(d))
+        spec = AddCostSpec(T, U, psi_container)
```

Comment:
Does `spec` need to be created in every loop iteration?  Or can one instance be shared?

Reply:
One instance can't be shared as different devices can have different services, so different terms in the `addtional_linear_terms` field